api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -515,16 +515,26 @@ private QueryJobConfiguration buildQueryJobConfigForDomain(
                           domain.toString())));
 
       // This adds the where clauses for cohorts and concept sets.
-      conceptSetSqlInClauseMaybe.ifPresent(
-          clause ->
-              queryBuilder
-                  .append(" WHERE \n(")
-                  .append(domainConceptIdInfo.getStandardConceptIdColumn())
-                  .append(clause)
-                  .append(" OR \n")
-                  .append(domainConceptIdInfo.getSourceConceptIdColumn())
-                  .append(clause)
-                  .append(")"));
+      if (Domain.SURVEY.equals(domain)) {
```

Comment:
Can we have it like :
   if (conceptSetSqlInClauseMaybe.isPresent()) {
        queryBuilder
            .append(" WHERE \n(")
            .append(domainConceptIdInfo.getStandardConceptIdColumn())
            .append(conceptSetSqlInClauseMaybe.get());
        if (Domain.SURVEY.equals(domain)) {
          queryBuilder.append(" OR \n")
              .append(domainConceptIdInfo.getSourceConceptIdColumn())
              .append(conceptSetSqlInClauseMaybe.get());
        }
        queryBuilder.append(")");

      }

To avoid duplication of code?

Reply:
Changed to remove any duplication.